### PR TITLE
Fix integration test for transactions in branch: feature/improve_transactions_processing_efficiency - Closes #3143

### LINKED ISSUE #3143: Fix integration test for transactions in branch: feature/improve_transactions_processing_efficiency
### Expected behavior
The integration tests for transactions should pass in the feature branch: `feature/improve_transactions_processing_efficiency`
### Actual behavior
The integration tests were failing because tests were not updated.
### Steps to reproduce
Run tests.
### Which version(s) does this affect? (Environment, OS, etc...)
feature/improve_transactions_processing_efficiency

### PR DESCRIPTION
**Pull request by SargeKhan**
### What was the problem?
The integration tests were failing for transactions after the removal of unconfirmed state.
### How did I fix it?
Fixed the integration tests for all transactions except for multisignature transactions. The tests for multisignature transactions will be fixed as part of #2825.
### How to test it?
Run integration tests

* The PR resolves #3143
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/test/mocha/integration/transactions/3_3_votes.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ describe('system test (type 3) - voting with duplicate submissions', () => {
 	let t = 0;
 
 	/* eslint-disable no-loop-func */
-	while (i < 30) {
+	while (i < 1) {
```
Comment (pablitovicente) [inline framework/test/mocha/integration/transactions/3_3_votes.js]: This should be 30 again I think?

File: framework/test/mocha/integration/transactions/0.0.address_collision.js [hunk 2]
```diff
@@ -173,9 +173,11 @@ describe('system test (type 0) - address collision', () => {
 									expect(res).to.be.undefined;
 									expect(err).to.be.not.null;
 									expect(err).to.equal(
-										`Invalid sender public key: ${publicKeys[1]} expected: ${
+										`Transaction: ${
+											secondTransactionWithData.id
+										} failed at .senderPublicKey: Invalid sender publicKey, actual: ${
 											publicKeys[0]
-										}`
+										}, expected: ${publicKeys[1]}`
```
Comment (mitsujutsu) [inline framework/test/mocha/integration/transactions/0.0.address_collision.js]: This test seems to fail

File: framework/src/modules/chain/logic/transaction_pool.js [hunk 3]
```diff
@@ -161,6 +163,31 @@ class TransactionPool {
 			...poolConfig,
 			...poolDependencies,
 		});
+
+		this.subscribeEvents();
```
[the remark this answers is not in the document]
Comment (SargeKhan) [reply framework/src/modules/chain/logic/transaction_pool.js]: We create a new instance of `pool` and the old instance is garbage collected, therefore we don't need to unsubscribeEvents.

File: framework/test/mocha/integration/transactions/0.0.address_collision.js (hunk 2 above)
Comment (SargeKhan) [reply framework/test/mocha/integration/transactions/0.0.address_collision.js]: Are you sure? It's passing on my machine.

Comment (mitsujutsu) [reply framework/test/mocha/integration/transactions/0.0.address_collision.js]: that's strange, it was failing on my machine yesterday

**Pull request merged**